operations/rack_resiliency/rack_resiliency.md

```diff
@@ -0,0 +1,551 @@
+have replicas and resiliency across racks.
+
+### 1. Categories of CSM critical services
+
+CSM services can be grouped into different categories based on the nature of support they provide to run the user jobs.
+
+#### 1.1 CSM platform services
+
+These services host the platform on which the CSM runs and provide the infra structure for running those services which are critical for user jobs.
+
+    * Example: BOS, BSS, CFS, gitea, IMS, iPXE etc.,
+
+#### 1.2. CSM critical services
+
+These services directly impact the way user jobs are executed and the availability of these services is very much essential for succeeding end user jobs.
+Critical services can not execute without the platform services. So there is dependency on the platform services for critical services to be healthy.
+
+    * Ex:  `slurm-ctld*`, `dns-unbound-manager` etc.,
+     
+### 2. Criterias for Critical services
+   
+* it should effect the workloads that are executed on the compute nodes
+    * Example: `slurmctld`, `slurm-operator`, `slurmdbd`, `pbs`, `pbs-comm` etc.,
+
+* it should provide the infra structure for executing jobs (storage, networking)
+    * Example:  `cray-ceph-csi-cephfs-provisioner`, `cray-ceph-csi-rbd-provisioner`, `coredns`, `cray-dns-unbound`, `sshot-net-operater` etc.,
+
+## RR solution overview
+
+The Rack Resiliency solution is implemented in 4 stages. The 4 stages are:
+
+* Management plane placement discovery and placement validation
+* Spreading of NCNs across Kubernetes zones, and spreading storage nodes across Ceph zones 
+* Distribution of critical services across zones
+* Monitoring failures and ensuring critical services are available across zones
+
+![Rack Resiliency solution overview](../../img/rack-resiliency-1.png)
+
+### 1 Discover management plane and validate for resiliency 
+
+#### 1.1 Identify racks and management nodes across racks
+
+The discovery of the physical racks and management nodes (Master, Worker and Storage) in Cray Ex system happens with the "Discovery" method, where during the node personalization stage of booting a management node, the discovery and validation operation is executed. The [Hardware State Manager (HSM)](../../glossary.md#hardware-state-manager-hsm) is used to get component details for segregating management nodes across racks in the form of [xnames](../../glossary.md#xname).
+
+The [System Layout Service (SLS)](../../glossary.md#system-layout-service-sls) is used to map these management node xnames to the corresponding Kubernetes and storage node names. This mapping of rack xnames to Kubernetes and storage node hostnames is stored in the below format as a JSON file to be consumed by the Kubernetes and Ceph zoning modules later.
+
+Example of JSON file containing rack to management NCN hostname mapping (`rr_hw_discovery.json`):
+
+```text
+{
+x3000:[ncn-m001,ncn-w001,ncn-w004,ncn-w007,ncn-s001],

```

Comment:
Is this the actual format of the file? If so, it isn't legal JSON anyway. Those strings would need to have quotes around them. We shouldn't call it a JSON file if it's not actually in JSON format. Or if it is actually in JSON format, the example should be corrected.